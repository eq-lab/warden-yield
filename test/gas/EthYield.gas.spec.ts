import { expect } from 'chai';
import { loadFixture } from '@nomicfoundation/hardhat-network-helpers';
import snapshotGasCost from '@uniswap/snapshot-gas-cost';
import { createEthYieldFork } from '../shared/fixtures';
import { ethers } from 'hardhat';
import { parseEther } from 'ethers';
import { USER_WARDEN_ADDRESS, setTokenBalance } from '../shared/utils';

describe('EthYield', () => {
  it('user stake, native', async () => {
    const { ethYield } = await loadFixture(createEthYieldFork);
    const [_, user] = await ethers.getSigners();

    const input = parseEther('1');
    await snapshotGasCost(
      Number(await ethYield.connect(user).stake.estimateGas(input, USER_WARDEN_ADDRESS, { value: input }))
    );
  });

  it('user stake, weth', async () => {
    const { weth9, ethYield } = await loadFixture(createEthYieldFork);
    const [_, user] = await ethers.getSigners();

    const input = parseEther('1');
    await setTokenBalance(await weth9.getAddress(), user.address, input);
    await weth9.connect(user).approve(ethYield.target, input);

    await snapshotGasCost(Number(await ethYield.connect(user).stake.estimateGas(input, USER_WARDEN_ADDRESS)));
  });
});
