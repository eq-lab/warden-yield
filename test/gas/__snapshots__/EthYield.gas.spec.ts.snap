// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EthYield gas reinit from Axelar, complete eigenLayer withdraw 1`] = `474897`;

exports[`EthYield gas reinit from evm, complete eigenLayer withdraw 1`] = `457519`;

exports[`EthYield gas reinit, complete lido withdraw 1`] = `832068`;

exports[`EthYield gas stake 1`] = `1072156`;

exports[`EthYield gas stake native, complete one eigenLayer and one lido withdraws 1`] = `1423026`;

exports[`EthYield gas stake with reinit 1`] = `859881`;

exports[`EthYield gas stake, complete lido withdraw 1`] = `1163606`;

exports[`EthYield gas unstake 1`] = `408688`;

exports[`EthYield gas unstake, complete eigenLayer withdraw 1`] = `704046`;

exports[`EthYield gas unstake, complete lido withdraw 1`] = `1011661`;

exports[`EthYield gas unstake, complete one eigenLayer and one lido withdraws 1`] = `1282681`;
